compiler/rustc_metadata/src/rmeta/decoder/cstore_impl.rs

```diff
@@ -561,6 +561,23 @@ pub(in crate::rmeta) fn provide(providers: &mut Providers) {
             tcx.untracked().cstore.freeze();
             tcx.arena.alloc_from_iter(CStore::from_tcx(tcx).iter_crate_data().map(|(cnum, _)| cnum))
         },
+        visible_crates: |tcx, ()| {
+            // The list of loaded crates is now frozen in query cache,
+            // so make sure cstore is not mutably accessed from here on.
+            tcx.untracked().cstore.freeze();
+
+            // if `#![feature(rustc_private)]` is enabled in the current crate, include private
+            // dependencies (e.g. of `std`.)
+            if tcx.features().enabled(sym::rustc_private) {
+                tcx.crates(())
+            } else {
+                tcx.arena.alloc_from_iter(
+                    CStore::from_tcx(tcx)
+                        .iter_crate_data()
+                        .filter_map(|(cnum, meta)| (!meta.private_dep).then_some(cnum)),
```

Comment:
can u just make this a `filter` then `map`? i think `(!bool).then_some` is so unreadable lol

Reply:
like, i spent a few minutes going "why is this `private_dep.then_some()` before seeing that it was `(!private_dep).then_some()`

Reply:
Query is all gone